.github/workflows/lint.yml

```diff
@@ -1,41 +1,48 @@
-name: Lint
-# Lint runs golangci-lint over the entire cosmos-sdk repository
-# This workflow is run on every pull request and push to main
-# The `golangci` will pass without running if no *.{go, mod, sum} files have been changed.
+name: golangci-lint
 on:
-  pull_request:
   push:
+    tags:
+      - v*
     branches:
+      - master
       - main
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
 jobs:
   golangci:
-    name: golangci-lint
+    name: lint
     runs-on: ubuntu-latest
     steps:
       - uses: actions/setup-go@v3
         with:
           go-version: 1.18
-      - uses: technote-space/get-diff-action@v6.0.1
-        id: git_diff
-        with:
-          PATTERNS: |
-            **/**.go
-            go.mod
-            go.sum
```

Comment:
we usually opt to have this to not run this position of ci when there are no changes to the go code.

Reply:
I recommend that we run it all the time since the compute to do that costs nothing and the previous setup resulted in not linting at all.

Reply:
But there's no sense in running it if there are zero code changes.

Reply:
Yeah I can roll with it, I’ll make that change I’m just worried that I could introduce another bug that reproduces the prior state

Reply:
please revert this change, after this we can look at merging